lua/lean/infoview.lua

```diff
@@ -280,8 +280,17 @@ end
 -- Close the infoview associated with the current window.
 function M.close()
   if not M.is_open() then return end
+  local infoview = M._teardown(get_idx())
```

Comment:
What's being done here seems to be identical to what `close_win()` was doing previously - looks like this was an attempt to fix window closing/toggling? See latest commit.

Reply:
Don't quite get it yet - feel free to have a go at it and maybe I'll understand once I see the code.